dolphin/alert_manager/trap_receiver.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_log import log
+
+from pysnmp.entity import engine, config
+from pysnmp.carrier.asyncore.dgram import udp
+from pysnmp.entity.rfc3413 import ntfrcv
+from pysnmp.proto.api import v2c
+from pysnmp.smi import builder, view, rfc1902, error
+
+from dolphin.alert_manager import constants
+
+LOG = log.getLogger(__name__)
+
+# Currently static mib file list is loaded, logic to be changed to load all mib file
+MIB_LOAD_LIST = ['SNMPv2-MIB','IF_MIB']
+
+class TrapReceiver(object):
+    """Trap listening and processing functions"""
+
+    def __init__(self, trap_receiver_address, trap_receiver_port,
+                 snmp_mib_path, mib_view_controller=None, snmp_engine=None):
+        self.mib_view_controller = mib_view_controller
+        self.snmp_engine = snmp_engine
+        self.trap_receiver_address = trap_receiver_address
+        self.trap_receiver_port = trap_receiver_port
+        self.snmp_mib_path = snmp_mib_path
+
+    def _mib_builder(self):
+        """Loads given set of mib files from given path."""
+        mib_builder = builder.MibBuilder()
+        try:
+            self.mib_view_controller = view.MibViewController(mib_builder)
+
+            # set mib path to mib_builder object and load mibs
+            mib_path = builder.DirMibSource(self.snmp_mib_path),
+            mib_builder.setMibSources(*mib_path)
+            if len(MIB_LOAD_LIST) > 0:
+                mib_builder.loadModules(*MIB_LOAD_LIST)
+        except error.MibNotFoundError:
```

Comment:
Is this exception need to be throw out to the upper level?

Reply:
Ok, updated